FlipViceFeed/app/src/main/java/com/joelimyx/flipvicefeed/main/main/MainAdapter.java

```diff
@@ -0,0 +1,164 @@
+            }
+        });
+        animateRecyclerView(holder.mArticleItemLayout,position);
+    }
+
+    @Override
+    public int getItemCount() {
+        return mArticleList.size();
+    }
+
+    /*---------------------------------------------------------------------------------
+    // Helper Method
+    ---------------------------------------------------------------------------------*/
+    public void swapData(String path){
+        String url = "http://vice.com/api/getlatest/category/"+path;
+        mArticleList.clear();
+
+        StringRequest request = new StringRequest(Request.Method.GET, url,
+                new Response.Listener<String>() {
+                    @Override
+                    public void onResponse(String response) {
+
+                        //Extracting data
+                        GsonArticle gsonArticle = new Gson().fromJson(response,GsonArticle.class);
+                        List<Item> items = gsonArticle.getData().getItems();
+
+                        //Setup up adapter to recycler view
+                        mArticleList.addAll(items);
+                        notifyDataSetChanged();
+                    }
+                },
+                new Response.ErrorListener() {
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        Toast.makeText(mContext, "Error getting articles", Toast.LENGTH_SHORT).show();
+                    }
+                });
+        VolleySingleton.getInstance(mContext).addToRequestQueue(request);
+    }
+
+    public void addData(List<Item> addedItems){
+        int start = getItemCount();
+        mArticleList.addAll(addedItems);
+        notifyItemRangeInserted(start,addedItems.size());
+    }
+
+    public void addQueriedData(List<Item> addedTags) {
+        mArticleList.clear();
+        mArticleList.addAll(addedTags);
+        notifyDataSetChanged();
+    }
+
+    private void animateRecyclerView(View viewToAnimate, int position) {
+        if (position > lastPosition) {
+            Animation animation = AnimationUtils.loadAnimation(mContext, R.anim.list_item_main_animation);
+            viewToAnimate.startAnimation(animation);
+            lastPosition = position;
+        }
+    }
+
+    public void showSearchResults(String query) {
+        ArrayList<Item> matchingList = new ArrayList<>();
+        for (int i = 0; i < mArticleList.size(); i++) {
+            for (String tag:mArticleList.get(i).getTags()) {
+                if (tag.toLowerCase().equals(query.toLowerCase())){
+                    matchingList.add(mArticleList.get(i));
+                }
+
+            }
+        }
+        if (!matchingList.isEmpty()) {
+            Intent intent = new Intent(mContext, SearchActivity.class);
+            intent.putParcelableArrayListExtra(SearchActivity.SEARCH_KEY, matchingList);
+            mContext.startActivity(intent);
```

Comment:
This feels a little strange to be launching a new activity from the adapter (as opposed to inside an OnClickListener set on a ViewHolder). It works, but your adapter shouldn't have to make navigation decisions. Try to separate your concerns more.